tests/test_addon_detail.py

```diff
@@ -19,6 +20,19 @@ def test_extension_meta_card(selenium, base_url, variables):
     assert addon.summary.is_displayed()
 
 
+@pytest.mark.nondestructive
+def test_detail_author_links(selenium, base_url, variables):
+    extension = variables['detail_extension_slug']
+    selenium.get(f'{base_url}/addon/{extension}')
+    # read the add-on author name and clicks on it
+    addon = Detail(selenium, base_url)
```

Comment:
Could we verify that this page opens? We could just use the ```wait_for_page_to_load``` function since that will return the page's respective object.

Reply:
@jrbenny35 is `wait_for_page_to_load` a selenium built in function then? I mean, there is no need for me to define a separate `WebDriverWait` function in the page objects class (the Detail class in this example)?

If so, I would simply need to add the following line to the above test?

```
addon = Detail(selenium, base_url)
addon.wait_for_page_to_load()
```

Reply:
@jrbenny35 so, if I understand this correctly, I should make sure that all page objects have a `wait_for_page_to_load` function defined, which waits for a particular element to be visible or clickable etc. Then, when that specific page is used in a test, I need to remember to also call that function, as you exemplified above.

For the Details page, I already have a custom `wait_for_page_to_load` function, so I just need to call it in the test as well:

```
    def wait_for_page_to_load(self):
        self.wait.until(
            expected.invisibility_of_element_located(
                (By.CLASS_NAME, 'LoadingText')))
```
This particular wait makes sure that all page sections are loaded, i.e. there are no loading elements still present.

Reply:
Yes exactly! This just allows for us to use a page wherever/whenever we want and we can  be sure it is loaded before doing anything.